src/unit_tests/shared/test_enrollment_op.c

```diff
@@ -337,6 +359,31 @@ void test_w_enrollment_concat_group(void **state) {
     w_enrollment_concat_group(buf, group);
     assert_string_equal(buf, " G:'EXAMPLE_GROUP'");
 }
+/**********************************************/
+/************* w_enrollment_concat_key ****************/
+void test_w_enrollment_concat_key_empty_buff(void **state) {
+    expect_assert_failure(w_enrollment_concat_key(NULL, (keyentry *)1));
```

Comment:
I don´t suggest using this kind of pointer cast. It´s easier to understand if we declare a keyentry structure and pass its direction.

Reply:
Changed!